OsmAnd/src/net/osmand/plus/views/PointLocationLayer.java

```diff
@@ -53,38 +56,48 @@ private void initUI() {
 
 		area = new Paint();
 		area.setColor(view.getResources().getColor(R.color.pos_area));
-		
+
 		aroundArea = new Paint();
 		aroundArea.setColor(view.getResources().getColor(R.color.pos_around));
 		aroundArea.setStyle(Style.STROKE);
 		aroundArea.setStrokeWidth(1);
 		aroundArea.setAntiAlias(true);
-		
-		updateIcons(view.getSettings().getApplicationMode(), false);
+
 		locationProvider = view.getApplication().getLocationProvider();
+		updateIcons(view.getSettings().getApplicationMode(), false, isLocationOutdated());
 	}
-	
+
 	@Override
 	public void initLayer(OsmandMapTileView view) {
 		this.view = view;
 		initUI();
 	}
 
+	private boolean isLocationOutdated() {
+		if (locationProvider.getLastKnownLocation() != null) {
+			return System.currentTimeMillis() - locationProvider.getLastKnownLocation().getTime() >
```

Comment:
Do we need to call it every time ? System.currentTimeMillis() ? Practically 60 times a second.